src/dbspro/cli/correctfastq.py

```diff
@@ -2,64 +2,63 @@
 Combines starcode output files into raw fastq files creating error corrected fastq files
 """
 import logging
-
 from tqdm import tqdm
-
-import dbspro.utils as dbspro
-
+import dnaio
+from collections import Counter
 
 logger = logging.getLogger(__name__)
 
 
 def main(args):
 
-    logger.info("Starting analysis")
-    logger.info("Processing file " + args.err_corr)
-    generator = dbspro.FileReader(args.err_corr)
+    logger.info(f"Starting analysis")
+    logger.info(f"Processing file: {args.err_corr}")
+
     err_corr = dict()
-    for line in tqdm(generator.fileReader()):
+    clusters = set()
+    with open(args.err_corr, "r") as file:
+        for line in tqdm(file):
+            try:
+                cluster_seq, num_reads, raw_seqs_list = line.split()
+            except ValueError:
+                logging.warning(f"Non-default starcode output line: {line}")
+                continue
+            clusters.add(cluster_seq)
+            for raw_seq in raw_seqs_list.split(","):
+                if raw_seq not in err_corr:
+                    err_corr[raw_seq] = cluster_seq
 
-        import time
-        try: cluster_seq, num_reads, raw_seqs_list = line.split()
-        except ValueError:
-            print("WARNING! Non-default starcode output line:\t'" + line + "'")
-        for raw_seq in raw_seqs_list.split(","):
-            if not raw_seq in err_corr:
-                err_corr[raw_seq] = cluster_seq
+    logger.info(f"Clusters: {len(clusters)}")
 
-    generator.close()
-    logger.info("Error corrected sequenced parsed.")
+    logger.info(f"Error corrected sequenced parsed.")
 
-    logger.info("Correcting sequences and writing to output file.")
-    generator = dbspro.FileReader(args.raw_fastq)
-    no_err_corr_seq = int()
-    tot_reads = int()
-    corr_seqs = int()
-    with open(args.corr_fastq, 'w') as openout:
-        for read in tqdm(generator.fastqReader()):
+    logger.info(f"Correcting sequences and writing to output file.")
 
-            tot_reads += 1
-            if read.seq in err_corr:
-                cluster_seq = err_corr[read.seq]
-                read.seq = cluster_seq
-                openout.write(read.fastq_string())
-                corr_seqs += 1
+    counter = Counter()
+    with dnaio.open(args.raw_fastq, mode="r", fileformat="fastq") as reader, \
+         dnaio.open(args.corr_fastq, mode="w", fileformat="fastq") as openout:
+        for read in tqdm(reader):
+
+            counter['tot_reads'] += 1
+            if read.sequence in err_corr:
+                read.sequence = err_corr[read.sequence]
+
+                # FIX: Possibly the sequence of the error corrected read is different from the original
+                if len(read.qualities) != len(read.sequence):
+                    read.qualities = "G"*len(read.sequence)
```

Comment:
Doesn't really matter. Could also change quality but we could also just write fasta as out.

Reply:
We talked about this and decided to leave it for now and make the file into a fasta instead in a later PR. I will make a new issue with this.